File: Angular_DotNet/PortalGalleto/src/app/model/GalletaViewModel.ts
import { MedidaViewModel } from "./MedidaViewModel";
import { RecetaViewModel } from "./RecetaViewModel";

export class GalletaViewModel{
    id: number;
    nombre: string | null;
    cantidad: number;
    precioVenta: number;
    precioProduccion: number;
    imagen: string | null;
    pesoGalleta: number;
    idMedida: number;
    medida: MedidaViewModel | null;

    recetamodel: RecetaViewModel[];

    constructor(val: GalletaViewModel){
        this.id = val.id ? val.id : 0;
        this.nombre = val.nombre ? val.nombre : null;
        this.cantidad = val.cantidad ? val.cantidad : 0;
        this.precioVenta = val.precioVenta ? val.precioVenta : 0;
        this.precioProduccion = val.precioProduccion ? val.precioProduccion : 0;
        this.imagen = val.imagen ? val.imagen : null;
        this.pesoGalleta = val.pesoGalleta ? val.pesoGalleta : 0;
        this.idMedida = val.idMedida ? val.idMedida : 0;
        this.medida = val.medida ? val.medida : null;

        this.recetamodel = [];
    }
}

File: Angular_DotNet/ServiciosGalleto/Servicios.Dominio/Servicios.Entidad/ViewModel/GalletaViewModel.cs
using Servicios.Entidad.Model;
using System;
using System.Collections.Generic;
using System.Text;

namespace Servicios.Entidad.ViewModel
{
    public class GalletaViewModel
    {
        public int? id { get; set; }
        public string? nombre { get; set; }
        public decimal? cantidad { get; set; }
        public decimal? precioVenta { get; set; }
        public decimal? precioProduccion { get; set; }
        public string? imagen { get; set; }
        public decimal? pesoGalleta { get; set; }
        public int? idMedida { get; set; }
        public MedidaViewModel? medida { get; set; }

        public List<RecetaViewModel>? recetamodel { get; set; }
    }
}
